cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -2,485 +2,166 @@
+    unit = request.GET.get("unit", None)
+    parent = request.GET.get("parent", None)
 
-    if isinstance(course_key, LibraryLocator):
-        courselike_module = modulestore().get_library(course_key)
-        context = {
-            'context_library': courselike_module,
-            'courselike_home_url': reverse_library_url("library_handler", course_key),
-            'library': True
-        }
+    courselike_key = CourseKey.from_string(course_key_string)
+    library = isinstance(courselike_key, LibraryLocator)
+    if library:
+        successful_url = reverse_library_url("library_handler", courselike_key)
+        courselike_module = modulestore().get_library(courselike_key)
+        context_name = "context_library"
     else:
-        courselike_module = modulestore().get_course(course_key)
-        context = {
-            'context_course': courselike_module,
-            'courselike_home_url': reverse_course_url("course_handler", course_key),
-            'library': False
-        }
+        successful_url = reverse_course_url("course_handler", courselike_key)
+        courselike_module = modulestore().get_course(courselike_key)
+        context_name = "context_course"
 
-    context['export_url'] = export_url + '?_accept=application/x-tgz'
+    if not has_course_author_access(request.user, courselike_key):
+        raise PermissionDenied()
 
-    # an _accept URL parameter will be preferred over HTTP_ACCEPT in the header.
-    requested_format = request.REQUEST.get('_accept', request.META.get('HTTP_ACCEPT', 'text/html'))
+    export_url = reverse(
+        "course_import_export_handler",
+        kwargs={
+            "course_key_string": unicode(courselike_key),
+        }
+    ) + "?accept=application/x-tgz"
+
+    export_url += "&{0}".format(
+        urlencode({
+            "redirect": reverse_course_url(
+                "export_handler",
+                unicode(courselike_key)
+            )
+        })
+    )
 
-    if 'application/x-tgz' in requested_format:
+    if parent:
         try:
-            tarball = create_export_tarball(courselike_module, course_key, context)
-        except SerializationError:
-            return render_to_response('export.html', context)
-        return send_tarball(tarball)
-
-    elif 'text/html' in requested_format:
-        return render_to_response('export.html', context)
+            parent = CourseKey.from_string(parent)
+            edit_unit_url = reverse_usage_url("container_handler", parent.location)
+        except (InvalidKeyError, AttributeError):
+            return render_to_response("export.html", {
+                context_name: courselike_module,
+                "export_url": export_url,
+                "raw_err_msg": _(
+                    "Invalid parent supplied; \"{supplied_key}\" is not a "
+                    "valid course or library key."
+                ).format(supplied_key=parent),
+                "library": library
+            })
```

Comment:
Yes, I think this message should probably be logged, as this should never happen while trying to export courses by clicking through the UI. I'll add that in.